include/libtorrent/torrent_peer.hpp

```diff
@@ -269,7 +285,7 @@ namespace libtorrent {
 
 		bool operator()(torrent_peer const* lhs, torrent_peer const* rhs) const
 		{
-#if TORRENT_USE_I2P
+#if TORRENT_USE_I2P || TORRENT_USE_RTC
```

Comment:
this looks suspicious. The code enabled by this preprocessor check is purely i2p-related

Reply:
Indeed there should be a comparison of `is_rtc_addr` flags.